cognite/neat/graph/extractors/_dexpi.py

```diff
@@ -0,0 +1,202 @@
+            for attribute, value_definitions in generic_attributes.items():
+                predicate = as_neat_compliant_uri(attribute)
+                for value_definition in value_definitions:
+                    if literal := cls._value_definition2literal(value_definition):
+                        triples.append((id_, predicate, literal))
+
+        return triples
+
+    @classmethod
+    def _value_definition2literal(cls, definition: dict) -> Literal | None:
+        if "Value" not in definition or "Format" not in definition:
+            return None
+
+        # case: when language is present we create add language tag to the literal
+        elif "Language" in definition and "Value" in definition:
+            return Literal(definition["Value"], lang=definition["Language"])
+
+        # case: when ValueURI is present we use it instead of Value
+        # this would be candidate for ENUMs in CDF
+        elif "ValueURI" in definition:
+            return Literal(definition["ValueURI"], datatype=XSD[definition["Format"]])
+
+        # case: when Format is not string we make sure to add the datatype
+        elif definition["Format"].lower() != "string":
+            return Literal(definition["Value"], datatype=XSD[definition["Format"]])
+
+        # case: when Format is string we add the literal without datatype (easier to read triples, less noise)
+        else:
+            return Literal(definition["Value"])
+
+    @classmethod
+    def _get_element_label(cls, element: Element) -> str | None:
+        if children := get_children(element, "Label", 1):
+            if grandchildren := get_children(children[0], "Text", 1):
+                if "String" in grandchildren[0].attrib:
+                    return grandchildren[0].attrib["String"]
+
+        # extension for schema version 3.3, where text is used to "label" without a <label> parent
+        elif children := get_children(element, "Text", 1):
+            if "String" in children[0].attrib:
+                return children[0].attrib["String"]
+
+        return None
+
+    @classmethod
+    def _get_element_generic_attributes(cls, element: Element) -> dict:
+        # TODO: This requires more work as there are multiple groupings of GenericAttributes
+
+        attributes = {}
+        if children := get_children(element, "GenericAttributes", 1):
+            if grandchildren := get_children(children[0], "GenericAttribute"):
+                for generic_attribute in grandchildren:
+                    # extension for schema version 3.3, where "AttributeURI" is not included
+                    if "AttributeURI" in generic_attribute.attrib:
+                        if generic_attribute.attrib["AttributeURI"] not in attributes:
+                            attributes[generic_attribute.attrib["AttributeURI"]] = [generic_attribute.attrib]
+
+                        else:
+                            attributes[generic_attribute.attrib["AttributeURI"]].append(generic_attribute.attrib)
```

Comment:
Good case for `defaultdict(list)`?
```suggestion
        attributes = defaultdict(list)
        if children := get_children(element, "GenericAttributes", 1):
            if grandchildren := get_children(children[0], "GenericAttribute"):
                for generic_attribute in grandchildren:
                    # extension for schema version 3.3, where "AttributeURI" is not included
                    if "AttributeURI" in generic_attribute.attrib:
                        attributes[generic_attribute.attrib["AttributeURI"]].append(generic_attribute.attrib)

        return dict(attributes)
```

Reply:
yeps :)